Xperia 10 III fixes

sledges opened the pull request:
Local build itself was untested.

llewelld reviewed (approved):
I also didn't try a local build with these changes, but they look entirely sane to me.

I'm happy to test the local build if that's considered necessary, but it will take a bit of time to complete.

llewelld commented:
I don't see any obvious changes. Just rebased? Either way, still approved.

sledges commented:
> I don't see any obvious changes. Just rebased? Either way, still approved.

Click on "force-pushed" link above to see the diff. Added link to upstream bug tracker.
